### ISSUE TITLE
DigiFinex may take your funds without clearly explaining you why, part 3

### ISSUE THREAD

**Issue by JovianMoon** [ccxt/ccxt#7066]
So, I have been running two market making bots on Digifinex for over a year now, with quite decent profits. The bots were trading different coin pairs, not doing arbitrage, not trading with each other nor collaborating with anybody else's bots. Both used CCXT for the communication and had rate limiting enabled. 

I had also made many successful withdrawals from the account of the other bot. With the second bot account I hadn't ever tried to withdraw, I think. When I finally tried to withdraw with this bot in the beginning of May, it led to the withdrawal getting stuck to a "Pending for confirmation" state. I tried to withdraw with the other and same thing happened. For some reason I was able to make one smaller withdrawal later.

I then contacted the Digifinex support and asked if they can do something about the stuck withdrawals. This led to both accounts being disabled (couldn't place order with the API, couldn't log in through their website). Many emails were send back and forth after this, and I even got one response via email from @oliverdigifinex, who has been helping people with with disabled accounts in the previous threads of similar nature here on github. Ultimately, it seemed that they had made up their mind and were not going to let me have my accounts back. Here are couple of quotes of how they described it:

- "Your account is blocked and cannot be unblocked, and the assets in the account cannot be taken out."
- "Our staff has confirmed your account many times and the account cannot be unblocked"
- "Sorry, the account has been determined that the restriction cannot be lifted, please use another account"

And here are the best reasons I was able to get from them as for Why did they block my accounts:

- "DigiFinex received reports of profit violations, we verified the account issues, and confirmed account violations using APIs to obtain benefits, so the account was banned"
- "Please do not profit through any violations, thank you"
- "Your blocked account, what operation did you make when using the API, and the action is not allowed by the project side"
- "Before the block, some API behaviors are not allowed to appear. For specific behaviors, you can take a look at the API settings"
- "This is the punishment you need to accept in violation of the API"

I tried to ask them many times what the API/profit violations were, but they never responded with any more detailed answers than these. Confusingly, at some point after I explained them basic principles of how my bots worked, they responded:

- "I believe most of your account operations are normal
It's just that you pay attention to the blocked account."

After this it was back to their usual not very detailed responses about the account that cannot be unblocked and about API violations. To this date my accounts are still disabled with the profits I have made in them.

For what it is worth, I say (and said to them also) that I didn't conduct any API/profit violations on purpose. I was just trying to run market making bots in order to make profit. There is a possibility that I have broken their "Quantitative Indicators" listed in their API documentation, as I hadn't noticed them. However, it is quite hard to confirm this afterwards and they have also not indicated anything like that in their responses. They never sent any emails prior to closing the account asking to fix problems with the API usage. The accounts were simply disabled in cold blood. There could have been some API bans for breaking their API rules, but this too is hard to confirm afterwards.

Why is DigiFinex behaving like this? The explanations that come to my mind are:
1) They are too incompetent to determine if the behavior of their users accounts are legitimate.
2) This is a malicious act from their part. They don't want people making profits in their platform and keep withdrawing them.
3) My bots actually did something so horrifyingly wrong, that my accounts deserved to be disabled and have their funds taken.

What ever the situation, I think the user should be given a detailed description of what was done wrong and given an chance to explain the situation in order to recover their funds. And a warning messages Before disabling any accounts would feel like a good thing to have.

I'm writing this here, because I hope it would help to recover my accounts and funds as some people were able to do in previous threads of the same kind. Also, judging mine and the previous cases, it would seem to me that it is not safe for anyone to do API trading on the DigiFinex exchange. Previously it has helped some people to write to these kind of threads as communication with their support hasn't worked out. But who knows how many people have permanently lost their funds because they didn't do so? Unless DigiFinex comes up with just about perfect explanation for their behavior, or it turns out that I have actually done something very bad, I too would recommend to have DigiFinex API removed from the CCXT library.

@kroitor If you wish to have my full discussion with their support, I can share the emails to you privately.

@oliverdigifinex @oscaroyk Feel free to explain here why your users deserve to have their funds taken without warning messages or without given a chance to defend themselves because you don't tell them what exactly did they do wrong. Or maybe even explain and give proof for the API/profit violations that I have done (check support messages with id 324932202, the messages are signed as "Jovian").

If anyone else has had similar situations with DigiFinex recently, it would be nice to hear from you also.

**Comment by oscaroyk** [comment]
Dear @JovianMoon 

Please be advised your API have violated the API manipulation rules as set out in https://docs.digifinex.com/en-ww/v3/ a critcal annex to the terms of use.

Pursuant to regulation 3 of the API rules, both your bot accounts have in tandem received numerous warnings for high frequency placing and cancellation of orders leading to a violation of the anti-manipulation rules. The first and second bans amount to 30 mins, with the 3rd a 24 hours ban and subsequently, a universal ban followed by a wipe. That is to say the account is fully wiped and irretrievable.  

While these are drastic measures they are taken against accounts and high frequency trading and market manipulation that make it unsafe or threaten the users and community on the exchange.

We hope that you exercise caution with the implementation of bots and observe the anti-manipulation rules for the safe trading of the community. All users are responsible for the actions of their bots.

Kind Regards
DigiFinex

**Comment by kroitor** [comment]
Thank you for a thorough report @JovianMoon !

> The bots were trading different coin pairs, not doing arbitrage, not trading with each other nor collaborating with anybody else's bots. 

↑ I don't see anything wrong with any of the above. Market-making, high-frequency trading, arbitrage and collaborating with other bots within the rate limits – are all fully legit ways of cryptocurrency speculation with the exchange markets to me.

> They never sent any emails prior to closing the account asking to fix problems with the API usage. The accounts were simply disabled in cold blood.

Personally, I don't think blocking the funds or locking out the user accounts is acceptable in any way. The funds must remain accessible. If the usage somehow violates exchanges' rules, the funds should not be blocked. Instead, the exchange should warn you, but should still allow you to withdraw whatever you've had to the date, and just restrict further usage of the exchange, but don't withhold the funds and should not wipe the account.

> What ever the situation, I think the user should be given a detailed description of what was done wrong and given an chance to explain the situation in order to recover their funds. And a warning messages Before disabling any accounts would feel like a good thing to have.

I agree with that.

> But who knows how many people have permanently lost their funds because they didn't do so?

It looks like you're not alone having the same difficulties. We've had similar reports here: 

- https://github.com/ccxt/ccxt/issues/7066
- https://github.com/ccxt/ccxt/issues/6424

> Unless DigiFinex comes up with just about perfect explanation for their behavior, or it turns out that I have actually done something very bad, I too would recommend to have DigiFinex API removed from the CCXT library.

I would agree to that as well. 

---------

@oliverdigifinex, @oscaroyk – can you help resolve this issue in a way that would not block anyone else in the future without a clearly stated reason and a proper warning notification by email? I believe @JovianMoon has to be refunded.

Ideally – switch off the blocking, until you fix it on your side. It seems to be false-flagging and blocking users for legit trading activities. The main concern here is the inability to access the funds after a lock. I would understand if you want to keep the blocking functionality, however, the system should not lock the funds post-fact after allowing the users to trade for a year, on an attempt to withdraw funds.

Doing that without a proper warning and withholding users' funds is not very far from stealing in broad daylight. This is not the reputation you would want for DigiFinex. We will be happy to keep having DigiFinex listed in CCXT, if DigiFinex helps resolve this issue positively (not just this specific one, but all further issues of the same nature).

If the user violates provision 3, just disallow placing further orders for that specific user, but don't lock the funds – that is nonsense. Also the system should warn the user clearly over email **beforehand**, not afterwards.

Otherwise we will have to consider delisting it, which we don't want, but will have to, if it keeps behaving this way. High-frequency trading isn't a crime. It's in fact the very opposite – automated HFT is the future of trading. If you don't agree – the system should lock the order placing for the user, but should not block the entire account from signing in and should still allow the user to withdraw his funds. If you don't want any sort of HFT – remove your API entirely and keep the trading via the website only. Obviously, our goal is to protect CCXT users from unreasonable damage and risks, hope for your understanding.

**Comment by oliverdigifinex** [comment]
The following is just my personal opinion, does not represent the attitude of DIgifinex, these two accounts directly do not have transactions, but there are hundreds of repeatedly registered accounts, they manipulate market prices through self-Tradling, conduct fraudulent transactions, these two accounts  is closely related to their time and transactions. We do not provide a detailed explanation because we are not willing to disclose our risk control algorithm. I respect that CCXT protects users, but what about organized crime?

**Comment by oliverdigifinex** [comment]
I will not participate in the handling of this case, my personal attitude will affect the fair results, @oscar will follow up

**Comment by silert** [comment]
Dear @oscaroyk 
My account #419892380 was also blocked 2 month ago and I received similar explanation from Oliver today by email, that I'm illegally manipulating the price of DFT.

You are wrong, because:
1. I use exactly the same algorithm on many exchanges and nobody blocked me except Digifinex and CoinTiger. 
Surely I didn't manipulate the price of DFT. I executed not so many orders and their size was relatively small to change the price.
2. I didn't get any noticeable profit on Digifinex during this year, so your budget shouldn't be affected. You can check my deposits and withdrawals, I deposited much more money than I have withdrawn.
3. It's hard to monitor bots when you run them on many exchanges in parallel. Sometimes orders get placed and cancelled in a loop due to simple mistake in the script and we notice it too late (I saw it many times in my bots).

At least should allow to withdraw our initial funds and block only API trading. 

Regarding multiple accounts, you can enforce passing KYC for all users, I think it should help.

**Comment by kroitor** [comment]
@oliverdigifinex 

> I respect that CCXT protects users, but what about organized crime?

Organized crime is not acceptible either, but the crime (if any) has to be proven clearly beyond  doubts. Otherwise making allegations of criminal activities against a user who was wiped without proofs – is not in favor of the exchange reputation. If you just leave it like that – that is definitely not it the interest of DigiFinex. True organized criminals usually don't complain.

> manipulate market prices through self-Tradling

If two different accounts were allowed to place two opposite orders and they matched – that is not a crime, it's a totally legit trade, paying fees to the exchange in the process. The exchanges usually implement self-trading prevention.

**Comment by JovianMoon** [comment]
Well @oliverdigifinex I'm certainly not the one registering hundreds of accounts nor collaborating anybody that is doing so. These two bots are the only ones that I was running or your exchange. Have you considered the possibility that when you have plenty of competing bots from multiple users it might sometimes end up looking like they are doing self trading between them even though they are actually working independently? Again, I have not been trying to do anything like that. Also not part of any crime organization if you can believe that.

**Comment by oliverdigifinex** [comment]
I don't think can try it in nasdaq  that useing different account to  manipulate price

**Comment by oliverdigifinex** [comment]
I just express my own opinion, @oscar will handle your case

**Comment by oliverdigifinex** [comment]
@silert，Are you @cryptoAcc0unt in #6424？

**Comment by cryptoAcc0unt** [comment]
Not sure why I was mentioned here, but I am not silert. For the record Digifinex unfroze me soon after my initial post and I have not had any issues since.

**Comment by kroitor** [comment]
@oliverdigifinex 

> https://www.nasdaq.com/articles/5-market-manipulation-tactics-and-how-avoid-them-2018-04-11

Market manipulation is a very specific term. It usually involves the following:

1. Fake News (not this case)
2. Pump And Dump (a derivative of fake news, not this case)
3. Spoofing The Tape (placing orders in the market with no intention of having them filled, obviously, not this case)
4. Wash Trading (a big player buys and sells the same security continually and nearly instantaneously – to boost the volumes and attract investors to the coin – also not the case, since you would want to delist the coin or at least address the coin team, not ban the user).
5. Bear Raiding (a large player places large sell orders to dump the price and hit stoplosses, also not this case)

Can you elaborate which specific _"price manipulation"_ do you mean? I think, none of the above from NASDAQ is violated in this issue. Not in this case at least. 

If you suspect there was some market manipulation – can you share the details? I don't see the exchange support having explained a specific reason for locking the funds. Moreover, again, it requires some proofs, otherwise any market-maker can be blamed in an attempt to "manipulate the price". Which is a totally legit activity – all trades manipulate the price. Market manipulation is illegal, but I don't think it has anything to do with price manipulations. It is a very slippery ground, since not all manipulation is criminal.

So, far it does not explain the blocking of funds in this case.

**Comment by JovianMoon** [comment]
I would also love to see some proof.

**Comment by oliverdigifinex** [comment]
> Dear @oscaroyk
> My account #419892380 was also blocked 2 month ago and I received similar explanation from Oliver today by email, that I'm illegally manipulating the price of DFT.
> 
> You are wrong, because:
> 
> 1. I use exactly the same algorithm on many exchanges and nobody blocked me except Digifinex and CoinTiger.
>    Surely I didn't manipulate the price of DFT. I executed not so many orders and their size was relatively small to change the price.
> 2. I didn't get any noticeable profit on Digifinex during this year, so your budget shouldn't be affected. You can check my deposits and withdrawals, I deposited much more money than I have withdrawn.
> 3. It's hard to monitor bots when you run them on many exchanges in parallel. Sometimes orders get placed and cancelled in a loop due to simple mistake in the script and we notice it too late (I saw it many times in my bots).
> 
> At least should allow to withdraw our initial funds and block only API trading.
> 
> Regarding multiple accounts, you can enforce passing KYC for all users, I think it should help.

In your email, I  deal with your issue on February 4th? I'm a little confused

**Comment by silert** [comment]
@oliverdigifinex That was another story. You blocked me on January 18 due to arbitrage trading. Then you unblocked me in February after the post #6424 explaining that arbitrage opportunities happened due to an issue on your side and you have to take all profit which I made (in fact you took ~1.2x more). So I continued trading hoping to return these losses and you blocked me again in April.

**Comment by binares** [comment]
I'm not saying this is the case here, but in my experience the smaller exchanges only care about "market manipulation" when the exchange itself is trading on their behalf, but realizes losses. E.g they employ their own market making bot, and due to misconfiguration (small spread, ob lag to global price) they experience significant losses due to their users arbitraging. That includes "blind" arbitrage from market making, as ~X spread still fills into orders that delay-deviate from global price.

**Comment by oliverdigifinex** [comment]
> @oliverdigifinex
> 
> > https://www.nasdaq.com/articles/5-market-manipulation-tactics-and-how-avoid-them-2018-04-11
> 
> Market manipulation is a very specific term. It usually involves the following:
> 
> 1. Fake News (not this case)
> 2. Pump And Dump (a derivative of fake news, not this case)
> 3. Spoofing The Tape (placing orders in the market with no intention of having them filled, obviously, not this case)
> 4. Wash Trading (a big player buys and sells the same security continually and nearly instantaneously – to boost the volumes and attract investors to the coin – also not the case, since you would want to delist the asset or at least address the coin team, not ban the user).
> 5. Bear Raiding (a large player places large sell orders to dump the price and hit stoplosses, also not this case)
> 
> Can you elaborate which specific _"price manipulation"_ do you mean? I think, none of the above from NASDAQ is violated in this issue. Not in this case at least.
> 
> If you suspect there was some market manipulation – can you share the details? I don't see the exchange support having explained a specific reason for locking the funds. Moreover, again, it requires some proofs, otherwise any market-maker can be blamed in an attempt to "manipulate the price". Which is a totally legit activity – all traders manipulate the price. Market manipulation is illegal, but I don't think it has anything to do with price manipulations. It is a very slippery ground, since not all manipulation is criminal.
> 
> So, far it does not explain the blocking of funds in this case.



> I'm not saying this is the case here, but in my experience the smaller exchanges only care about "market manipulation" when the exchange itself is trading on their behalf, but realizes losses. E.g they employ their own market making bot, and due to misconfiguration (small spread, ob lag to global price) they experience significant losses due to their users arbitraging. That includes "blind" arbitrage from market making, as ~X spread still fills into orders that delay-deviate from global price.

I said earlier that this does not happen with currencies traded globally，We also don’t make markets for small coins because we don’t want to hold small coins

**Comment by oliverdigifinex** [comment]
> I'm not saying this is the case here, but in my experience the smaller exchanges only care about "market manipulation" when the exchange itself is trading on their behalf, but realizes losses. E.g they employ their own market making bot, and due to misconfiguration (small spread, ob lag to global price) they experience significant losses due to their users arbitraging. That includes "blind" arbitrage from market making, as ~X spread still fills into orders that delay-deviate from global price.

I said earlier that this does not happen with currencies traded globally，We also don’t make markets for small coins because we don’t want to hold small coins

**Comment by oliverdigifinex** [comment]
> @oliverdigifinex That was another story. You blocked me on January 18 due to arbitrage trading. Then you unblocked me in February after the post #6424 explaining that arbitrage opportunities happened due to an issue on your side and you have to take all profit which I made (in fact you took ~1.2x more). So I continued trading hoping to return these losses and you blocked me again in April.

 May I ask do you share IP with 290798？OR it is another account of you?

**Comment by oliverdigifinex** [comment]
> @oliverdigifinex
> 
> > https://www.nasdaq.com/articles/5-market-manipulation-tactics-and-how-avoid-them-2018-04-11
> 
> 市场操纵是一个非常具体的术语。它通常涉及以下内容：
> 
> 1. 假新闻（不是这种情况）
> 2. Pump And Dump（假新闻的衍生产品，不是这种情况）
> 3. 欺骗磁带（显然，并非在这种情况下将订单放置在市场上，无意将其填充）
> 4. Wash Trading（大型参与者连续几乎立即购买和出售相同的证券–以增加交易量并吸引投资者购买代币-也不是这种情况，因为您想将资产除牌或至少与代币团队联系，而不是禁止用户）。
> 5. 熊袭击（大型玩家下达大量卖单以甩出价格并触及止损，在这种情况下也不是）
> 
> 您能详细说明一下具体的_“价格操纵”_吗？我认为，在此问题上，纳斯达克的上述所有规定均未被违反。至少不是这种情况。
> 
> 如果您怀疑有人操纵市场–您可以分享细节吗？我看不到交易所的支持人员解释了锁定资金的具体原因。而且，它再次需要一些证明，否则，任何市场做市商都可能被指责为试图“操纵价格”。这是完全合法的活动–所有交易者都在操纵价格。市场操纵是非法的，但我认为这与价格操纵无关。这是非常湿滑的地面，因为并非所有操纵都是犯罪行为。
> 
> 因此，到目前为止，这还不能解释这种情况下的资金阻塞。

I am not a trader, but as far as I know it is similar to this https://www.sec.gov/news/press-release/2014-229

**Comment by kroitor** [comment]
@oliverdigifinex to my mind that release from SEC is about manipulating the closing prices of a market – that is only applicable to conventional exchanges that have "open working hours", a schedule (like NySE and other conventional markets).

Such a manipulation is not applicable to a 24/7 cryptocurrency exchange like DigiFinex, though. And of course it still requires at least some proofs, otherwise it looks like the exchange cannot explain the exact reason for locking the funds. 

Regarding criminal activities, a lock of funds on behalf of the exchange may be considered a fraud, that is intentionally locking the funds of the users for the purposes of withholding their profits to the benefit of the exchange owners, without explaining the exact true reasons for doing so. That in itself would be a far more serious allegation towards the exchange itself, because direct fraud can be a more serious crime. 

I believe, nobody really wants to go into those waters, and it would be in the best interest of the exchange to acknowledge a mistake, refund the user, and optionally apologize for the inconvenience. This is what I would do, if I were an exchange and if I banned a user by my mistake. In that case the exchange will gain a better reputation of a service that does not let down their users. So, the exchange might even win a better image if it decides to do things right. The users will become much more confident in your service, and will continue using it.

However, if you choose the opposite and insist on the fault of the user in this case – that would most likely damage the reputation of the exchange if you don't want to show any proofs or specific reasons. In that regard, the Internet remembers everything and never forgets.

It is up to the exchange to decide the best way out of this situation. I'm just expressing my personal opinion.

**Comment by oliverdigifinex** [comment]
Once again, I’m just discussing my personal views. It does not mean the official attitude of digifinex. I always resolve problems quickly and don’t mind admitting mistakes, but I have my views on this kind of case. In addition, freezing is the most severe warning. We have never captured it. The user’s money, in addition to the rollback transaction, is refunded to the lost customer

**Comment by oliverdigifinex** [comment]
> @oliverdigifinex to my mind that release from SEC is about manipulating the closing prices of a market – that is only applicable to conventional exchanges that have "open working hours", a schedule (like NySE and other conventional markets).
> 
> Such a manipulation is not applicable to a 24/7 cryptocurrency exchange like DigiFinex, though. And of course it still requires at least some proofs, otherwise it looks like the exchange cannot explain the exact reason for locking the funds.
> 
> Regarding criminal activities, a lock of funds on behalf of the exchange may be considered a fraud, that is intentionally locking the funds of the users for the purposes of withholding their profits to the benefit of the exchange owners, without explaining the exact true reasons for doing so. That in itself would be a far more serious allegation towards the exchange itself, because direct fraud can be a more serious crime.
> 
> I believe, nobody really wants to go into those waters, and it would be in the best interest of the exchange to acknowledge a mistake, refund the user, and optionally apologize for the inconvenience. This is what I would do, if I were an exchange and if I banned a user by my mistake. In that case the exchange will gain a better reputation of a service that does not let down their users. So, the exchange might even win a better image if it decides to do things right. The users will become much more confident in your service, and will continue using it.
> 
> However, if you choose the opposite and insist on the fault of the user in this case – that would most likely damage the reputation of the exchange if you don't want to show any proofs or specific reasons. In that regard, the Internet remembers everything and never forgets.
> 
> It is up to the exchange to decide the best way out of this situation. I'm just expressing my personal opinion.

Their behaver is more like this https://www.sec.gov/news/press-release/2019-216，I don't mean @JovianMoon.

**Comment by oliverdigifinex** [comment]
> 在我看来，@ oliverdigifinex从SEC释放是操纵市场的收盘价-仅适用于具有“开放工作时间”，时间表的常规交易所（例如NySE和其他常规市场）。
> 
> 但是，这种操作不适用于像DigiFinex这样的24/7加密货币交易所。当然，它仍然至少需要一些证明，否则交易所似乎无法解释锁定资金的确切原因。
> 
> 关于犯罪活动，代表交易所锁定资金可能被视为欺诈，是有意锁定用户的资金，目的是为了将其利润代扣给交易所所有者，而没有解释其确切的真实原因。这样做。这本身将是对交易所本身的更为严重的指控，因为直接欺诈可能是更严重的犯罪。
> 
> 我相信，没有人真的愿意涉足这些领域，并且承认错误，退还用户以及对由此带来的不便表示歉意符合交易所的最大利益。如果我是交易所，并且由于错误而禁止了某个用户，那么我将这样做。在那种情况下，交换将获得不降低用户满意度的服务的更好声誉。因此，如果交易所决定做正确的事情，它甚至可以赢得更好的形象。用户将对您的服务更有信心，并将继续使用它。
> 
> 但是，如果您选择相反的做法并在这种情况下坚持用户的过错–如果您不想出示任何证据或特定原因，很可能会损害交易所的声誉。在这方面，互联网会记住一切，永远不会忘记。
> 
> 由交易所决定如何摆脱这种情况的最佳方法。我只是表达我的个人意见。

They behave more like this https://www.sec.gov/news/press-release/2019-216 ,I don't mean @JovianMoon

**Comment by kroitor** [comment]
> Once again, I’m just discussing my personal views. It does not mean the official attitude of digifinex.

I understand that. CCXT is trying to be in neutral non-judging position in general. We're under a MIT license, therefore it's up to the user to assess risks before entering the exchange. However, obviously, we would not want anyone to lose their money without a clear reason for locking, otherwise nobody would use this lib.

> I always resolve problems quickly and don’t mind admitting mistakes, 

We appreciate your help and involvement in resolving similar issues reported previously.

> but I have my views on this kind of case.

If you are a member of DigiFinex's team and you can shed the light on the exact cause of locking the author – that could really help resolve the issue.

**Comment by oliverdigifinex** [comment]
Digifinex will have an official attitude to deal with it. I only express my personal dissatisfaction with such cases. I always solve problems quickly and never mind admitting mistakes, but for such harvesters, we will see. In addition, we have never seized user funds, unless the rollback transaction is returned to the lost user, freezing is only the most severe warning

**Comment by oliverdigifinex** [comment]
> @oliverdigifinex
> 
> > Once again, I’m just discussing my personal views. It does not mean the official attitude of digifinex.
> 
> I understand that. CCXT is trying to be in neutral non-judging position in general. We're under a MIT license, therefore it's up to the user to assess risks before entering the exchange. However, obviously, we would not want anyone to lose their money without a clear reason for locking, otherwise nobody would use this lib.
> 
> > I always resolve problems quickly and don’t mind admitting mistakes,
> 
> We appreciate your help and involvement in resolving similar issues reported previously.
> 
> > but I have my views on this kind of case.
> 
> If you are a member of DigiFinex's team and you can shed the light on the exact cause of locking the author – that could really help resolve the issue.

@oscaroyk will support @JovianMoon and @silert

**Comment by kroitor** [comment]
@JovianMoon , @silert @oscaroyk, @oliverdigifinex guys, do you have a followup on this issue? If it is resolved, we would close it. Let us know if you have further comments or details.

**Comment by JovianMoon** [comment]
@kroitor Both of my accounts are still disabled. I emailed them on Wednesday and asked for an update. Oliver said he would ask this from someone. I also emailed to Oscar on Thursday with the email address I received from Oliver. I haven't got any messages at any point from Oscar nor any kind of update to my case. I hope they haven't decided that silence is the best solution to this issue...

**Comment by silert** [comment]
@kroitor I have the same situation. I sent email to Oscar and didn't get any response, the account is still disabled.

**Comment by kroitor** [comment]
@JovianMoon, @silert  thanks for letting us know! Sorry to hear that the problem is still there.

@oscaroyk @oliverdigifinex what is your comment on the above? DigiFinex has to react if you're interested in keeping the listed status. Either provide clear reasons or restore the accounts and the access to the users' funds. A week is more than enough to investigate and resolve it. If your stance on this issue is negative or indifferent, we will have to consider delisting it next week to protect other users from similar troubles in the future. I believe a positive resolution is in the best interest of DigiFinex in the first place.

**Comment by oscaroyk** [comment]
@kroitor Appreciate all of your involvement in this case. I believe we are sharing one of the common goals - Protect Users and maintain a fair trading environment. DigiFinex will NEVER take user funds without any reason or explanation, including all of the previous cases. 

There are several points I want to clarify here:

1. Either Github forum or personal email is not an official channel for DigiFinex CS services, all the reply in there may not be prompt. In DigiFinex, we offer 7x24x365 Customer Services through different official channels. If you have difficulty reaching out, please refer to below comprehensive instructions.  
https://digifinex.zendesk.com/hc/en-us/articles/360000525241

2. All suspended accounts will be received a corresponding warning or error code (refer in API docs) before the account has been frozen. Users have the responsibility to understand the rules and regulations they have accepted in Exchange. If there is anything remain unclear, please don't hesitate to contact our 7x24 CS services.

3. For any individual appeal case, user can send their UID, case details with justifications. Meanwhile, we will also provide an explanation about which rule you're violated or anything you can do to unfreeze your account.

4. For the above 1-2 specific cases, I would suggest sending an email to support@digifinex.com with your justifications/appeal reason, with CC to me for easy reference. I will speed up the case and give you a reply by this week. This is an exceptional handling case, if there is any similar case happened, please refer to the official CS channels I have mentioned. 

5. We're always striving the best to maintain a fair environment for all of the traders and projects. We will continue to review our internal process and enhance our communication effectiveness with users. 

Win-win situation is always the best way we are looking for, and delist is not an option for us, even I don't see there is any reason for this action while we're maintaining positive respond.  
I hope my answer above can settle all the concerns from you guys. @kroitor @silert @JovianMoon

**Comment by kroitor** [comment]
@silert @JovianMoon alright, i suggest one last try. Please, execute it precisely, literally to a word as noted in (4) written by @oscaroyk above. Send your UID, explicit case details and justifications to support@digifinex.com and CC @oscaroyk for easy reference. Send a duplicate of your appeal via (1).

Paste the exact datetimes of your appeals here in this issue when you send out the emails. From there, expect and wait for the answer from DigiFinex or from @oscaroyk within one week. 

We will decide further action at that time.

**Comment by JovianMoon** [comment]
@oscaroyk @kroitor Appeal email titled "Appeal to recover my disabled accounts" was sent to support@digifinex.com CC to Oscar email at Jun 15, 2020, 3:23 PM GMT+3.

@kroitor The other official channels mentioned in (1) are live chats via their app, website or their Telegram group. Would like me to start a live chat with them?

There is also one more option that is not mentioned at (1). It is this form, that can be accessed by clicking FeedBack on bottom of DigiFinex webpage: https://digifinex.zendesk.com/hc/en-us/requests/new . This was the method I first used for contacting them, and after that I received messages from support@digifinex.com and kept discussing with them. Would you like me to create a dublicate issue to their support with this method, @kroitor?

The problem with the UID:s is that I haven't written them down and cannot log back in to check what they are for my accounts. I may have been able to snatch UID for one of my accounts before they were both disabled, but I'm not sure which code I have anymore. I have been giving them this one number and the email addresses associated to my accounts while discussing with them and at no point have they been indicating that they Don't know what accounts I'm talking about. UID is also required to send that Zendesk form.

The reason I started this github issue and emailed those "personal" email addresses was that discussions with their support hadn't been going anywhere. I didn't want to try any live chats, because I had doubts it would help based on issues #7035 and #6424. I think there are only two messages that I have Only send to personal email addresses and not also to their support.

**Comment by frosty00** [comment]
@kroitor I fully support delisting Digifinex if @JovianMoon and @silert cases are not resolved. We can't have exchanges disabling our user's account as it will discourage them from trading with CCXT at all.

The market manipulation and links to the SEC as explanation as to why their accounts were disabled do not help Digifinex's case at all.

> Please be advised your API have violated the API manipulation rules as set out in https://docs.digifinex.com/en-ww/v3/ a critcal annex to the terms of use.

These kinds of rules go against the spirit of CCXT. Either this rule is removed from Digifinex, or Digifinex will get delisted from CCXT.

Consider this the final warning.

**Comment by kroitor** [comment]
@JovianMoon 

> @kroitor The other official channels mentioned in (1) are live chats via their app, website or their Telegram group. Would like me to start a live chat with them?

Yes, send the copied contents of your email appeal via the live chat. Also, send it to DigiFinex's Telegram group, or ask an admin there and pm the admin of the group if your appeal contains sensitive personal information.

> There is also one more option that is not mentioned at (1). It is this form, that can be accessed by clicking FeedBack on bottom of DigiFinex webpage: https://digifinex.zendesk.com/hc/en-us/requests/new .
> Would you like me to create a dublicate issue to their support with this method, @kroitor?

Yes, despite that this is redundant, still do send a duplicate via all means just to make sure.

> The problem with the UID:s is that I haven't written them down and cannot log back in to check what they are for my accounts.
> UID is also required to send that Zendesk form.

Use all means that you can possibly use. If some webform requires the UIDs that are unknown to you and you can't use the form because of that – skip it. You can try to find your UIDs in signup emails or whatever account history you might have, if any. 

If you can't find the UIDs – send as much info as you can that could help identify your case to DigiFinex, through all possible means. In other words, use all available communication channels to raise the chances for a positive resolution. If you can't use some of those channels due to impossible or unsatisfiable requirements – skip those.

----------

@frosty00 

I agree. I'm just trying to make sure that DigiFinex has a chance to resolve it positively. This is why I've asked the two banned users to make one final attempt.

----------

@oscaroyk 

The win-win situation is to:

- either provide clear explanations beyond reasonable doubts for banning the users
- or restore the access to the users' funds (i'd recommend this)

A reasonable response should show clear proofs on how, when, why and what has been violated. Not all responses are reasonable. Active market-making is not an illegal activity, definitely not a reason for locking the funds.

As exchange customers ourselves we know how frustrating it can be when the exchange does not react timely and adequately, footballing a customer from one channel to another. Sending the same templated answers over and over again in an endless loop is not a way out of this situation. A quality customer support service must help and support customers, and it should not take weeks or months.

**Comment by JovianMoon** [comment]
@kroitor As of now, Jun 15, 2020, 23:21 PM GMT+3, I have also sent the same message to them via their Zendesk (with both of my emails) and with their live chat in their website (both emails). I will tackle more with their Telegram group when there's an admin around, hopefully tomorrow.

And just to warn anyone trying to get help from their official English Telegram group: The first thing that happened to me after joining the group was that some scammer pretending to be from DigiFinex started PM:ing and asking 0.04 BTC to get the accounts back. Luckily I didn't fall for that...

**Comment by kroitor** [comment]
@JovianMoon alright, plz, keep us updated on the progress here.

**Comment by oscaroyk** [comment]
We have received users' email and will reply individually as their situation are different and also contained a lot of personal information. No need for any further action in our live chat/telegram group.

**Comment by JovianMoon** [comment]
@kroitor Got a rather promising sounding reply from their support today. Here's a quote:

- "In order to strike a balance between Exchange, Users and Projects, your account is confirmed to unfroze on or before 19 June 2020, after official announced by our customer service team."

The reason for the freezing of the accounts was "repeated violation of API trade provisions". There are three timestamps listed there as for when these violations occurred for both accounts, but there is no further explanations what the violations were. Apparently there were also user complaints. 

I'm not sure how much more I should show their response here, not that there is too much more to add. Let's see what they will say in their announcement. 

Currently both of the accounts are still disabled, hopefully that changes tomorrow.

**Comment by silert** [comment]
I've received similar response from their support and my account has been already unblocked. Thank you @kroitor, @oscaroyk and @oliverdigifinex for your help!

However it's still not clear which API trading rules have been violated and how to avoid account disabling in the future.

**Comment by oscaroyk** [comment]
@silert @JovianMoon Thank you for your patience and information to assist the investigation, I believe the above cases have been replied and solved, please don't hesitate to contact our 7x24 customer service team if you need any further support. 

DigiFinex will strike our best to provide a fair and secure trading platform to every user, meanwhile, we will continuously review all the rules and regulations from time to time, and reference with any market professions if appliable. We will also enhance our communication effectiveness and notification system to make sure a better communication with all of our users.

Again, Thank you for your continued support and your understanding is highly appreciated.

**Comment by kroitor** [comment]
@oscaroyk that's great news! Thank you so much! We appreciate your support!

@JovianMoon we stand by for your final confirmation here to get this issue resolved and closed. Please, let us know as soon as possible.

**Comment by JovianMoon** [comment]
@kroitor My accounts have been enabled, I was able to log back in, the funds were still there and my withdrawals were successful! 

@oscaroyk Thank you for coming around and restoring my accounts. I will continue bot trading on DigiFinex next week after making some adjustments in order to avoid breaking any API rules. Hopefully it will go more smoothly this time around. Your decision to enhance communication and notification system sounds like the right way to go.

@kroitor Thank you and CCXT for taking good care of your customers and being very professional throughout the process. I'm grateful that your library exist and look forward to utilizing it even more in the future.

I will close this issue after 24 hours has passed, unless there is still more discussion going on then.

**Comment by JovianMoon** [comment]
@kroitor Noticed this on DigiFinex announcements, I think it requires a change to the rate limits on the CCXT side: https://digifinex.zendesk.com/hc/en-us/articles/900001329403-Announcement-on-Adjustments-of-API-Performance-and-Usage-Rules

**Comment by kroitor** [comment]
@JovianMoon we will add those changes shortly, thx! Since both reported problems here are confirmed as resolved I'd close this issue, if you don't mind. Feel free to reopen it or just ask further questions, if any.

**Comment by AnastasiyaVoronova** [comment]
Hi, all!
Unfortunately, I am user of Digifinex and it was my first attempt to take money from my account (I made verification of my ID card, two-step verification with phone number, and so on).
I wanted to sell 1000 USDT and received answer from customer service, that to confirm that it's me, who makes this operation, I have to pay 1200 USDT to my account, and after that my transaction will be done.
Before it, they froze my 1000 USDT and I couldn't receive them back, neither sell it.

Obviously, I don't have 1200 USDT for such 'confirmation'.

Maybe someone could help me:

unfroze my account without paying 1200 USDT
get back my 1000 USDT
thank you all in advance! Any advice will be helpful!

**Comment by CompleteCorp** [comment]
www.completecorp.com.au is undertaking a Recovery Investigation into this company.

If you have been a victim, please feel free to register your details with us without obligation at https://www.completecorp.com.au/report-crypto-scam

**Comment by oscaroyk** [comment]
> hi , 3 days ago they locked my account with huge balance and open positions , no answer from email or live chat .
> 
> how should i deal with this?

can you provide more information for this case, or send email to support@digifinex.com for further assistant

**Comment by Simplenecro22** [comment]
> > hi , 3 days ago they locked my account with huge balance and open positions , no answer from email or live chat .
> > 
> > how should i deal with this?
> 
> Could you provide more information on this case, or send an email to support@digifinex.com for further assistance.

I already sent emails to support@digifinex.com , all they say in 4 days is to wait !
I had huge open positions :)
Are you sure digifinex is legit and regulated?

**Comment by Simplenecro22** [comment]
Just ask about my case from digifinex if you are a representative.

**Comment by Simplenecro22** [comment]
> > hi , 3 days ago they locked my account with huge balance and open positions , no answer from email or live chat .
> > 
> > how should i deal with this?
> 
> Could you provide more information on this case, or send an email to support@digifinex.com for further assistance.

I sent email , after 4 days still no explanation about why is that happening

**Comment by Simplenecro22** [comment]
> > > hi , 3 days ago they locked my account with huge balance and open positions , no answer from email or live chat .
> > > 
> > > how should i deal with this?
> > 
> > Could you provide more information on this case, or send an email to support@digifinex.com for further assistance.
> 
> I sent email , after 4 days still no explanation about why is that happening 

After 4 days they emailed me and asked for a video of me holding my ID card , i sent them and waiting for their response .

**Comment by Simplenecro22** [comment]
> > > > hi , 3 days ago they locked my account with huge balance and open positions , no answer from email or live chat .
> > > > 
> > > > how should i deal with this?
> > > 
> > > Could you provide more information on this case, or send an email to support@digifinex.com for further assistance.
> > 
> > I sent email , after 4 days still no explanation about why is that happening 
> 
> After 4 days they emailed me and asked for a video of me holding my ID card , i sent them and waiting for their response .

Provided every information needed but still not opened my account

**Comment by Simplenecro22** [comment]
Still no updates
Stay away

**Comment by Simplenecro22** [comment]
> > hi , 3 days ago they locked my account with huge balance and open positions , no answer from email or live chat .
> > 
> > how should i deal with this?
> 
> Could you provide more information on this case, or send an email to support@digifinex.com for further assistance.

Hey why don't you open my account ?

**Comment by Simplenecro22** [comment]
Total scammers , stay away

**Comment by Simplenecro22** [comment]
@oscaroyk my account is still closed

**Comment by TonyVernon20** [comment]
There are a lot of untrue recommendations and impostors, its hard to tell who is legit. If you have been scammed by fake crypto investors, lost money to scammers, I refer you to contact (WISETECHHACK@ GMAIL .COM) they will surely help you out. Took me long to find them.
